packages/core/types/src/file/provider.ts

```diff
@@ -144,4 +145,14 @@ export interface IFileProvider {
   getPresignedUploadUrl?(
     fileData: ProviderGetPresignedUploadUrlDTO
   ): Promise<ProviderFileResultDTO>
+
+  /**
+   * Get the file contents as a readable stream.
+   */
+  getAsStream(fileData: ProviderGetFileDTO): Promise<Readable>
```

Comment:
What's the use-case for stream methods for reading? SInce `getPresignedUploadUrl` gives you a link, you can then load the content as a stream or not, the caller is in full control of it, so they seem unnecessary.

Maybe I am missing something though

Reply:
for programatic ingestion from stream, for example parsing a file incrementally with async iterators over stream and import. It seems weird from a programmatic point of view to get a presigned URL and then load it from there no?

Reply:
It's not that weird, and it's a bit more versatile as the response is serializable. I am fine with adding it btw, just want to understand if we already have a use-case for it.

Also, do we need both as stream and as buffer? Wouldn't as stream suffice? The caller then can turn it in a buffer if they wanted to. Again, just thinking if we can keep the file module a bit leaner in case we don't have a use-case for it yet, but not a blocker for sure

Reply:
I just noticed that I pasted the wrong method above, apologies for that :D

Reply:
The `getPresignedDownloadUrl` as you get a URL to a file that you can then stream as in the example above

Reply:
Why should I first generate a URL and then read it as a stream myself using fetch, if the same can be encapsulated using the existing API's?

- `s3`: Use the SDK to fetch the file as a stream. Why create a URL, which includes semi-slow operations like generating a signature.
- `local`: Directly read it from the disk. Why create a URL, then hit an endpoint and then read the file from there?

Infact, this approach seems the opposite of keeping things lean, as we are giving up on the providers ability to perform an operation and instead self implementing those operations in other parts of the codebase.